File: api_log/src/main/java/com/lin/api_log/ActivityAspect.java
/*
 * ActivityAspect      2016-03-04
 * Copyright (c) 2016 hujiang Co.Ltd. All right reserved(http://www.hujiang.com).
 * 
 */

package com.lin.api_log;

import android.util.Log;
import org.aspectj.lang.JoinPoint;
import org.aspectj.lang.annotation.*;



@Aspect
public class ActivityAspect {
    public  final String TAG = this.getClass().getSimpleName().toString();

    @After("execution(* android.support.v4.app.FragmentActivity.onResume(..))")
    public void onResumeMethod(JoinPoint joinPoint) throws Throwable {
        Log.e("lin","resume="+joinPoint.getThis());

    }

    @After("execution(* android.support.v4.app.FragmentActivity.onPause(..))")
    public void onPauseMethod(JoinPoint joinPoint) throws Throwable {
        Log.e("lin","pause="+joinPoint.getThis());
    }
    @Before("execution(* android.support.v4.app.FragmentActivity.onCreate(..))")
    public void beforeCreate(JoinPoint joinPoint) throws Throwable {
        Log.e("lin","onCreate="+joinPoint.getThis());
    }

}

File: api_log/src/main/java/com/lin/apiLog/ActivityAspect.kt
/*
 * ActivityAspect      2016-03-04
 * Copyright (c) 2016 hujiang Co.Ltd. All right reserved(http://www.hujiang.com).
 * 
 */

package com.lin.apiLog

import android.util.Log
import org.aspectj.lang.JoinPoint
import org.aspectj.lang.annotation.*


@Aspect
class ActivityAspect {
    val TAG = this.javaClass.simpleName.toString()

    @After("execution(* android.support.v4.app.FragmentActivity.onResume(..))")
    @Throws(Throwable::class)
    fun onResumeMethod(joinPoint: JoinPoint) {
        Log.e("lin", "resume=" + joinPoint.getThis())

    }

    @After("execution(* android.support.v4.app.FragmentActivity.onPause(..))")
    @Throws(Throwable::class)
    fun onPauseMethod(joinPoint: JoinPoint) {
        Log.e("lin", "pause=" + joinPoint.getThis())
    }

    @Before("execution(* android.support.v4.app.FragmentActivity.onCreate(..))")
    @Throws(Throwable::class)
    fun beforeCreate(joinPoint: JoinPoint) {
        Log.e("lin", "onCreate=" + joinPoint.getThis())
    }

}